contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c

```diff
@@ -196,11 +196,60 @@ TsqlFunctionConvert(TypeName *typename, Node *arg, Node *style, bool try, int lo
 		result = (Node *) makeFuncCall(TsqlSystemFuncName("babelfish_conv_helper_to_date"), args, COERCE_EXPLICIT_CALL, location);
 
 	else if (type_oid == TIMEOID)
-		result = (Node *) makeFuncCall(TsqlSystemFuncName("babelfish_conv_helper_to_time"), args, COERCE_EXPLICIT_CALL, location);
-
+	{
+		Node	   *helperFuncCall;
+		helperFuncCall = (Node *) makeFuncCall(TsqlSystemFuncName("babelfish_conv_helper_to_time"), lcons(makeIntConst(typmod, location), args), COERCE_EXPLICIT_CALL, location);
+		
+		/*
+		 * add a type cast on top of the CONVERT helper function
+		 * so typmod can be applied
+		 */
```

Comment:
Can be re-written as: `apply typmod cast on top CONVERT`

Reply:
existing comment looks simple and easy to understand, let's not change it.